libraries/AP_HAL_ChibiOS/hwdef/BOTWINGF405/hwdef.dat

```diff
@@ -0,0 +1,220 @@
+# hw definition file for BOTWING-F405 hardware
+# tested on the BOTWING-F405 board
+
+# MCU class and specific type
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID AP_HW_BOTWINGF405
+
+# crystal frequency
+OSCILLATOR_HZ 24000000
+
+define STM32_ST_USE_TIMER 5
+define CH_CFG_ST_RESOLUTION 32
+
+# reserve 16k for bootloader, 16k for OSD and 32k for flash storage
+FLASH_RESERVE_START_KB 64
+FLASH_SIZE_KB 1024
+
+define HAL_STORAGE_SIZE 15360
+STORAGE_FLASH_PAGE 1
+
+# only one I2C bus
+I2C_ORDER I2C1
+
+# only one I2C bus (BARO - DPS310)
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+
+# SPI1 bus for GYRO (ICM42688P)
+PA5 SPI1_SCK  SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+PA4 GYRO_CS   CS
+
+# SPI2 for FLASH (W25Q128)
+PB13 SPI2_SCK   SPI2
+PB14 SPI2_MISO  SPI2
+PB15 SPI2_MOSI  SPI2
+PB12 FLASH_CS   CS
+
+# SPI3 for OSD (AT7456E)
+PC10 SPI3_SCK  SPI3
+PC11 SPI3_MISO SPI3
+PB5  SPI3_MOSI SPI3
+PA15 OSD_CS    CS
+
+# For debugging
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+
+# LEDs
+define AP_NOTIFY_GPIO_LED_2_ENABLED 1
+PC0 LED_A OUTPUT LOW GPIO(0)
+PC5 LED_B OUTPUT LOW GPIO(1)
+
+define HAL_GPIO_A_LED_PIN 0
+define HAL_GPIO_B_LED_PIN 1
+
+# Buzzer
+PB0 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+
+# --------- Power Control GPIOs and Relays ---------
+
+# 9V Power Enable Control (PB1 → Relay1)
+PB1 NINE_V_ENABLE OUTPUT GPIO(81) LOW
+define RELAY1_PIN 81   # Relay1 controls 9V_ENABLE
+
+# Secondary Power/Enable Control (PC13 → Relay2)
+PC13 ENABLE_2 OUTPUT GPIO(82) LOW
+define RELAY2_PIN 82   # Relay2 controls ENABLE_2
+
+----------------------------------------------------
+
+# Analog pins
+PC1 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC2 RSSI_ADC_PIN      ADC1 SCALE(1)
+PC3 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+define HAL_BATT_VOLT_PIN 11
+define HAL_BATT_VOLT_SCALE 11.0
+
+define BOARD_RSSI_ANA_PIN 12
+
+define HAL_BATT_CURR_PIN 13
+define HAL_BATT_CURR_SCALE 37.0
+
+define HAL_BATT_MONITOR_DEFAULT 4
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 UART5 USART6
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# --------- SBUS, PPM or ELRS Configurable Block ---------
+
+# --- SBUS-INVERTED Mode ---
+# comment these lines for ELRS mode
```

Comment:
Comment can me remove - there is no need to comment, ALT will do the right thing